### ISSUE TITLE
Test "lostnumxv" fails with CR

### ISSUE THREAD

**Issue by kyrsjo** [SixTrack/SixTrack#169]
Both fort.90 and fort.10 are DIFF.

**Comment by kyrsjo** [comment]
Note this commit:
https://github.com/SixTrack/SixTrack/commit/8af413b17b4920a4f4ab3fc0302b0a7520d24b40
It's from PR #137

**Comment by JamesMolson** [comment]
a comment in sixtrack.s:

```
! If CR we have to worry about turns printed in fort.6
! If lost should be OK, otherwise we need to use nnuml instead
! of the numl in numxv/nnumxv???? Eric.
! where we reset [n]numxv to nnuml UNLESS particle lost
! Now we shall try using that fix at start of tracking
```

**Comment by kyrsjo** [comment]
I ran this test in version v4.5.14 (the oldest tag we currently have), with the same result - It works for the normal version, DIFF for cr version.

**Comment by JamesMolson** [comment]
Looking in the input file we have
```
TRACKING PARAMETERS-----------------------------------------------------
1000 0 30 2.500 6.500 0 1 0 0 50 50
1 1 0 1 2
0 0 1 1 1 50000 2
NEXT
```

Looking in the fort.90, tracking seems to stop after 50 turns, but up until that point tracking is the same.

What are the last 4 entries on line 1, the 0 0 50 50.

No other input fort.3 files seems to have these (grep ^TRAC SixTest/*/fort.3 -A1).
The manual says these are:
niu(1), niu(2)
numlcp
numlmax

but no description is provided.

Looking in sixtrack.s we have:
```
! Two new options for Turns to checkpoint and max turns/job.
      if(iclr.eq.1) read(ch1,*,round='nearest')                         &
     &numl,numlr,napx,amp(1),amp0,ird,imc,niu(1),niu(2),numlcp,numlmax
```
setting max turns/job to 50 when the canonical expects 1000 might cause problems.


Later in the code:
```
+if cr
!  does not call CRPOINT if restart=.true.
!  (and note that writebin does nothing if restart=.true.
          if(mod(numx,numlcp).eq.0) call callcrp()
          restart=.false.
```

and for numlmax
```
! and now reset numl to do only numlmax turns
      nnuml=min((numlcr/numlmax+1)*numlmax,numl)
      write (93,*) 'numlmax=',numlmax,' DO ',numlcr,nnuml
```
etc etc, so this test will only every do 50 turns in CR mode.


If I delete these last 4 entries, the CR run passes the test.

I'll let @kyrsjo check that what I am saying makes sense and put a fix in.

Would you also please fill in the missing section in the documentation for these input entries?

**Comment by rdemaria** [comment]
It looks like the extension to split a long simulation in multiple boinc
tasks

On 4 Feb 2017 12:30, "James Molson" <notifications@github.com> wrote:

Looking in the input file we have

TRACKING PARAMETERS-----------------------------------------------------
1000 0 30 2.500 6.500 0 1 0 0 50 50
1 1 0 1 2
0 0 1 1 1 50000 2
NEXT

Looking in the fort.90, tracking seems to stop after 50 turns, but up until
that point tracking is the same.

What are the last 4 entries on line 1, the 0 0 50 50.

No other input fort.3 files seems to have these (grep ^TRAC
SixTest/*/fort.3 -A1).
The manual says these are:
niu(1), niu(2)
numlcp
numlmax

but no description is provided.

Looking in sixtrack.s we have:

! Two new options for Turns to checkpoint and max turns/job.
      if(iclr.eq.1) read(ch1,*,round='nearest')                         &
     &numl,numlr,napx,amp(1),amp0,ird,imc,niu(1),niu(2),numlcp,numlmax

setting max turns/job to 50 when the canonical expects 1000 might cause
problems.

Later in the code:

+if cr
!  does not call CRPOINT if restart=.true.
!  (and note that writebin does nothing if restart=.true.
          if(mod(numx,numlcp).eq.0) call callcrp()
          restart=.false.

and for numlmax

! and now reset numl to do only numlmax turns
      nnuml=min((numlcr/numlmax+1)*numlmax,numl)
      write (93,*) 'numlmax=',numlmax,' DO ',numlcr,nnuml

etc etc, so this test will only every do 50 turns in CR mode.

If I delete these last 4 entries, the CR run passes the test.

I'll let @kyrsjo <https://github.com/kyrsjo> check that what I am saying
makes sense and put a fix in.

Would you also please fill in the missing section in the documentation for
these input entries?

—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub
<https://github.com/SixTrack/SixTrack/issues/169#issuecomment-277437470>,
or mute the thread

**Comment by kyrsjo** [comment]
So:
 * `numlcp` is the ammount of turns to do between every checkpointing. Default value is 1000.
 * `numx` is simply turn number (counting from 1) minus 1, i.e. the turn number counting from 0.
 * `numlmax` has default value 1'000'000; it seems to be used for stopping tracking after that many turns IN THE CR VERSION. The default value should probably be higher, I see no reason why we should artificially limit the ammount of turns artificially like this.

I agree, let's (1) document these (i can do that) and (2) remove the last and very stupid parameter...

**Comment by kyrsjo** [comment]
Great catch @JamesMolson 👍

**Comment by kyrsjo** [comment]
Fixed in #205
